models/Pokedex.go

```diff
@@ -0,0 +1,33 @@
+package models
+
+import (
+	"fmt"
+
+	"github.com/dannegm/academy-go-q12021/constants"
+	"github.com/dannegm/academy-go-q12021/helpers"
+)
+
+// Pokedex stores a map of Pokemon by ID
+type Pokedex map[int]Pokemon
+
+// PokedexFromFile return a list of Pokemon
+func PokedexFromFile() (pokes Pokedex, err error) {
+	rows, err := helpers.ReadFile(fmt.Sprintf("%s/pokemon.csv", constants.AssetsPath))
+
+	if err != nil {
+		return
+	}
+
+	pokes = make(Pokedex)
```

Comment:
If you know how many pokemon are there (len(rows)), you could initialize `pokes` with that value.

Internally, when making a slide with no size, it starts with size 1, then, when capacity is full, it reallocates to a new slide with the double of size, when this new slide is full, it again create a new slide with double the capacity and reallocates all data, and so on.